### PR TITLE
Remove eigh usage on mac 

### PR DESCRIPTION
**Pull request by Ericgig**
**Description**
On mac, with OpenBlas, `eigh` can cause segmentation fault, see #1197. For 4.5.1, we simply skipped the tests causing problems ( #1257). This replace the calls of `numpy.linalg.eigh` to `numpy.linalg.eig`, and lapack calls of `ZHEEVR` to `ZGEEV` when at risk of segfault. This will result in slower `Qobj.eigenstates` and `brmesolve`. 

Tests where adapted, the issues were the phase of the eigenvectors and the lower numerical precision of `eig`.

While the flag to use `eig` instead of `eigh` is `qutip.settings.eigh_unsafe`, it must be set before importing `qutip.sparse` to have any effect. Thus it is not a true settings, (like `has_openmp`).

**Changelog**
Replace calls of `eigh` to `eig` when at risk of segfault.

### CODE REVIEW

**Comment by nonhermitian** [comment]
FYI, One has to be a bit careful as there is no internal orthonormalization step in eig or the lapack equivalent. This is trouble for degenerate systems. 

Ran into for the same issue in Qiskit.

File: qutip/tests/test_measurement.py [hunk 1]
```diff
@@ -143,64 +156,37 @@ def test_measurement_statistics_input_errors():
...
-    np.random.seed(seed)
-    measurements = []
-    for _ in expected_measurements:
-        value, new_state = measure(op, state)
-        measurements.append((value, new_state))
-    assert_(measurements == expected_measurements)
+def check_measure(op, state):
+    """ Check that all measure results are in measurement_statistics which
+    was previously tested.
+    This remove the depandance on the prng.
```
Comment (nathanshammah) [inline qutip/tests/test_measurement.py:162]:
```suggestion
    This remove the dependance on the prng.
```

File: qutip/sparse.py [hunk 2]
```diff
@@ -51,7 +51,43 @@
...
+        val, vec = la.eig(mat)
+        val = np.real(val)
+        idx = np.argsort(val)
+        val = val[idx]
+        vec = vec[:, idx]
+        if eigvals:
+            val = val[eigvals[0]:eigvals[1]+1]
+            vec = vec[:, eigvals[0]:eigvals[1]+1]
+        same_eigv = 0
+        for i in range(1, len(val)):
```
Comment (Ericgig) [inline qutip/sparse.py:73]: I do the orthonormalization raise by Paul Nation in this loop.

**Comment by Ericgig** [comment]
May not be needed anymore with scipy 1.5? To test more before merging (#1299 )

**Comment by jakelishman** [comment]
I now get segfaults on Mac with scipy 1.5.  I think this PR is still needed.

**Comment by jakelishman** [comment]
While this isn't an alternative for this PR, scipy 1.5 made new LAPACK drivers available for `scipy.linalg.eigh`, one or more of which might not have the same problems in OpenBLAS on Mac.  Still, we can't require scipy 1.5 when that only released like 5 days ago, but it's interesting going forwards.

**Pull request merged**